modules/raft/src/integrationTest/java/org/apache/ignite/raft/server/ItJraftCounterServerTest.java

```diff
@@ -709,6 +733,85 @@ public void testTimerThreadsCount() {
                 "All timer threads: " + timerThreads.toString());
     }
 
+    /**
+     * The test shows that all committed updates are applied after a RAFT group restart automatically.
+     * Actual data be available to read from state storage (not a state machine) directly just after the RAFT node started.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testApplyUpdatesOutOfSnapshot() throws Exception {
```

Comment:
Is it possible to write a unit test instead of integration one?

Reply:
It might, but in the case, I will be needed to copy methods to start/stop raft nodes and raft clients.